velox/type/tests/VariantTest.cpp

```diff
@@ -291,3 +290,116 @@ TEST(VariantTest, opaqueSerializationNotRegistered) {
       std::make_shared<opaqueSerializationTestStruct>());
   EXPECT_THROW(opaqueAfterRegistration.serialize(), VeloxException);
 }
+
+TEST(VariantTest, toJsonRow) {
+  auto rowType = ROW({{"c0", DECIMAL(20, 3)}});
+  EXPECT_EQ(
+      "[123456.789]",
+      variant::row({static_cast<int128_t>(123456789)}).toJson(rowType));
+
+  rowType = ROW({{"c0", DECIMAL(10, 2)}});
+  EXPECT_EQ(
+      "[12345.67]",
+      variant::row({static_cast<int64_t>(1234567)}).toJson(rowType));
+
+  rowType = ROW({{"c0", DECIMAL(20, 1)}, {"c1", BOOLEAN()}, {"c3", VARCHAR()}});
+  EXPECT_EQ(
+      "[1234567890.1,true,\"test works fine\"]",
+      variant::row({static_cast<int128_t>(12345678901),
+                    variant((bool)true),
+                    variant((std::string) "test works fine")})
+          .toJson(rowType));
+
+  // Row variant negative tests
+  rowType = ROW({{"c0", DECIMAL(10, 3)}});
+  VELOX_ASSERT_THROW(
+      variant::row({static_cast<int128_t>(123456789)}).toJson(rowType),
+      "Wrong type in variant::toJson");
```

Comment:
Why is this wrong type? Perhaps, include the types in the message for clarity.

Reply:
Changed the messaged.